src/data/polynomial.lean

```diff
@@ -971,6 +977,128 @@ lemma ne_zero_of_monic (h : monic p) : p ≠ 0 :=
 
 end nonzero_comm_semiring
 
+section comm_semiring
+variables [comm_semiring α] [decidable_eq α] {p q : polynomial α}
+
+/-- `dix_X p` return a polynomial `q` such that `q * X + C (p.coeff 0) = p`.
+  It can be used in a semiring where the usual division algorithm is not possible -/
+def div_X (p : polynomial α) : polynomial α :=
+{ to_fun := λ n, p.coeff (n + 1),
```

Comment:
Isn't this just `finsupp.map_domain (+1)`?

Reply:
If you use that definition, I guess we need a simp-lemma `coeff_div_X` that contains most of the lines below in its proof. Either way, that simp-lemma seems to be a good idea.

Reply:
You are right, I guess we need a `map_domain` for injective functions.

Reply:
Do you want a separate definition of `map_domain` for injective functions, or just lemmas about how `map_domain` works with injective functions?

Reply:
I have the lemmas in a private repository. I can add them.

But maybe its better to provide a separate function which assumes injective functions (or maybe embeddings?)  Like the split in finset with `finset.map` and `finset.image`.

Reply:
Turns out this is not `map_domain (+1)`, nor is it `map_domain (-1)`, because that would add the `X^1` and `X^0` terms, whereas the desired behaviour is to throw away the `X^0` term.

Reply:
Is this ready to merge in that case?

Reply:
Ouch. Sorry for the confusion.
Yes, its ready for merge!